src/PublicApiAnalyzers/Core/Analyzers/DeclarePublicApiAnalyzer.cs

```diff
@@ -18,14 +16,8 @@ namespace Microsoft.CodeAnalysis.PublicApiAnalyzers
     [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
     public sealed partial class DeclarePublicApiAnalyzer : DiagnosticAnalyzer
     {
-        /// <summary>
-        /// Cache from additional text instance to the api data we have read out for that specific file.  We only store
-        /// data for additional texts that explicitly match the public/internal api file names we expect.
-        /// <para/>
-        /// Intentionally not static.  We don't want this table living globally.  Rather, it should be tied to the
-        /// lifetime of this type and whatever owner it has.
-        /// </summary>
-        private readonly ConditionalWeakTable<AdditionalText, ApiData> _additionalTextToApiData = new();
+        private static readonly SourceTextValueProvider<ApiData> s_shippingApiDataProvider = new(static text => ReadApiData(text, isShippedApi: true));
+        private static readonly SourceTextValueProvider<ApiData> s_nonShippingApiDataProvider = new(static text => ReadApiData(text, isShippedApi: false));
```

Comment:
i'd honestly prefer we just have AdditionalTextValueProvider

Reply:
IMO this is a solid move forward for the analyzer in the known absence of that API. Also note that this analyzer references Roslyn 1.2.1 for maximum compatibility with downstream projects, so even if AdditionalTextValueProvider was added, it would potentially be problematic to adopt in the short term.